gapis/api/transform2/transform_chain.go

```diff
@@ -157,23 +181,27 @@ func (chain *TransformChain) GetNextTransformedCommands(ctx context.Context) err
 		log.I(ctx, "Transforming... (%v:%v)", chain.currentCommandID, currentCommand)
 	}
 
-	err := chain.transformCommand(ctx, currentCommand, chain.currentCommandID)
+	inputCmds := make([]api.Cmd, 0)
+	inputCmds = append(inputCmds, currentCommand)
+	err := chain.transformCommands(ctx, chain.currentCommandID, inputCmds, 0)
 	if err != nil {
 		log.E(ctx, "Replay error (%v:%v): %v", chain.currentCommandID, currentCommand, err)
 	}
+
 	chain.currentCommandID++
 	return err
 }
 
-func (chain *TransformChain) handleInitialState(state *api.GlobalState) (*api.GlobalState, error) {
-	for _, t := range chain.transforms {
-		if t.RequiresAccurateState() {
-			// Melih TODO: Temporary check until we implement accurate state
-			panic("Implement accurate state")
-		}
+func (chain *TransformChain) stateMutator(ctx context.Context, id api.CmdID, cmds []api.Cmd) error {
+	beginTransformIndex := chain.currentTransformIndex + 1
```

Comment:
I'd add some comments making the saving and restoring of member variables to the call stack as a means of recursion explicit. It's very easy to miss reading this. It's a very short and simple looking function, but actually hides quite a lot of subtle complexity. Would be nice to document!

Reply:
Sure. But as parallel with the previous command, can we add it to the next PR. I am planning to open it today.